SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialLatencyTracer.cpp

```diff
@@ -487,41 +468,46 @@ TraceKey USpatialLatencyTracer::CreateNewTraceEntryRPC(const AActor* Actor, cons
 
 	if (UClass* ActorClass = Actor->GetClass())
 	{
-		if (const UFunction* Function = ActorClass->FindFunctionByName(*FunctionName))
+		switch (Type)
 		{
-			ActorFuncKey Key{ Actor, Function };
-			if (TrackingTraces.Find(Key) == nullptr)
+		case ETraceType::RPC:
+			if (const UFunction* Function = ActorClass->FindFunctionByName(*Target))
 			{
-				const TraceKey _TraceKey = GenerateNewTraceKey();
-				TrackingTraces.Add(Key, _TraceKey);
-				return _TraceKey;
+				ActorFuncKey Key{ Actor, Function };
```

Comment:
This seems to be duplicated and could be taken out into a function:

if (TrackingTraces.Find(Key) == nullptr)
{
    const TraceKey _TraceKey = GenerateNewTraceKey();
    TrackingTraces.Add(Key, _TraceKey);
    return _TraceKey;
}
UE_LOG(LogSpatialLatencyTracing, Warning, TEXT("(%s) : ActorFunc already exists for trace"), *WorkerId);

Reply:
They are using different maps so it's a bit awkward, it could be made into a template function but then it'd have to live in the header. Thoughts?